pkg/client/unit_test.go

```diff
@@ -59,21 +59,109 @@ func TestUnitUpdateWithNewMap(t *testing.T) {
 	require.NoError(t, err)
 
 	// This should return TriggeredConfigChange, as we have an actually new map
-	got := defaultTest.updateState(UnitStateHealthy, UnitLogLevelDebug, 0, nil, newUnit, 2)
+	got := defaultTest.updateState(UnitStateHealthy, UnitLogLevelDebug, 0, nil, newUnit, 2, nil)
 	assert.Equal(t, TriggeredConfigChange, got)
 }
 
 func TestUnitUpdateLog(t *testing.T) {
-	got := defaultTest.updateState(UnitStateHealthy, UnitLogLevelInfo, 0, nil, &proto.UnitExpectedConfig{}, 2)
+	got := defaultTest.updateState(UnitStateHealthy, UnitLogLevelInfo, 0, nil, &proto.UnitExpectedConfig{}, 2, nil)
 	assert.Equal(t, TriggeredLogLevelChange, got)
 }
 
 func TestUnitUpdateFeatureFlags(t *testing.T) {
-	got := defaultTest.updateState(UnitStateHealthy, UnitLogLevelInfo, 1, &proto.Features{}, &proto.UnitExpectedConfig{}, 2)
+	got := defaultTest.updateState(UnitStateHealthy, UnitLogLevelInfo, 1, &proto.Features{}, &proto.UnitExpectedConfig{}, 2, nil)
 	assert.Equal(t, TriggeredFeatureChange, got)
 }
 
 func TestUnitUpdateState(t *testing.T) {
-	got := defaultTest.updateState(UnitStateStopped, UnitLogLevelInfo, 1, &proto.Features{}, &proto.UnitExpectedConfig{}, 2)
+	got := defaultTest.updateState(UnitStateStopped, UnitLogLevelInfo, 1, &proto.Features{}, &proto.UnitExpectedConfig{}, 2, nil)
 	assert.Equal(t, TriggeredStateChange, got)
 }
+
+func TestUnitUpdateAPMConfig(t *testing.T) {
+
+	var initialUnitNoAPM = Unit{
+		expectedState: UnitStateHealthy,
+		logLevel:      UnitLogLevelDebug,
+		featuresIdx:   0,
+		features:      nil,
+		configIdx:     1,
+		config:        &proto.UnitExpectedConfig{},
+		apm:           nil,
+	}
+
+	notEmptyAPMCfg := &proto.APMConfig{
+		Elastic: &proto.ElasticAPM{
+			Environment: "test",
+			ApiKey:      "apikey",
+			SecretToken: "",
+			Hosts:       []string{"host1", "host2"},
+			Tls: &proto.ElasticAPMTLS{
+				SkipVerify: false,
+				ServerCert: "/path/to/server/cert",
+				ServerCa:   "/path/to/server/ca",
+			},
+		},
+	}
+
+	type testcase struct {
+		name         string
+		initialState Unit
+		update       struct {
+			//unitState   UnitState
+			//UnitLogLvl  UnitLogLevel
+			//featuresIdx uint64
+			//features    *proto.Features
+			//unitConfig  *proto.UnitExpectedConfig
+			//cfgIdx      uint64
```

Comment:
Did you intentionally leave these commented-out fields in?

Reply:
Ah this was a quick copy paste that I used to remind myself what could go in a unit object in case I needed to set multiple fields in the test. 
In the end I only needed the apm config so everything else remained commented, I will remove the extrase